zebedee-cms/src/main/java/com/github/onsdigital/zebedee/keyring/LegacyKeyringImpl.java

```diff
@@ -349,7 +350,18 @@ public void assignTo(User src, User target, List<CollectionDescription> assignme
 
         for (CollectionDescription desc : assignments) {
             if (!srcCache.keySet().contains(desc.getId())) {
-                throw new KeyringException(CACHED_KEY_MISSING_ERR, desc.getId());
+                // If a key is not in the src users keyring cache then there is a potential issue with how the keys
+                // are assigned. We'll log this error instead of throwing it so user functionality isn't affected but
+                // this should be investigated if it occurs unexpectedly.
+                error().collectionID(desc)
+                        .data("src_user", src.getEmail())
+                        .data("target_user", target.getEmail())
+                        .log("error assiging collection key to target user. The src user is missing this collection key" +
```

Comment:
Please can I ask what is the difference between creating the error in this way to throwing a new exception.

Reply:
`error()` creates new log event with level `Error`, it doesn't actually throw an exception.

Originally this was throwing an exception which was causing other stuff to break, so to reduce the wider impact on the rest of the service I've updated this to log the error instead of throwing it.

Reply:
Thank you, this makes sense